community/kernel/src/main/java/org/neo4j/kernel/api/txtracking/TransactionIdTracker.java

```diff
@@ -39,9 +40,11 @@
     private static final TimeUnit POLL_UNIT = TimeUnit.MILLISECONDS;
 
     private final TransactionIdStore transactionIdStore;
+    private final LongPredicate isAvailable;
 
-    public TransactionIdTracker( TransactionIdStore transactionIdStore )
+    public TransactionIdTracker( TransactionIdStore transactionIdStore, LongPredicate isAvailable )
```

Comment:
There is no easy way to do so. Since the `AvailabilityGuard` cannot be fetched from the dependency resolver.